src/pages/angular/virtual-scroll.md

```diff
@@ -0,0 +1,123 @@
+---
+title: 'Virtual Scroll in Angular'
+---
+
+In previous versions of Ionic, there had been a `virtual-scroll` directive to help with list virtualization. But as of recent, Angular provides it's own solution with the `@angular/cdk` package.
+
+## Install
+
+To install the CDK Scroller, first install `@angular/cdk`:
+
+```shell
+npm add @angular/cdk
+```
+
+This provides a collection of different utilities, but we'll focus on `ScrollingModule` for now.
+
+When we want to use the CDK Scroller, we'll need to import the module in our component. For example, in a tabs starter project, we can add our import to the `tabs1.module.ts` file.
+
+```diff
+  import { IonicModule } from '@ionic/angular';
+  import { NgModule } from '@angular/core';
+  import { CommonModule } from '@angular/common';
+  import { FormsModule } from '@angular/forms';
+  import { Tab1Page } from './tab1.page';
+  import { ExploreContainerComponentModule } from '../explore-container/explore-container.module';
++ import { ScrollingModule } from '@angular/cdk/scrolling'
+  import { Tab1PageRoutingModule } from './tab1-routing.module';
+
+  @NgModule({
+    imports: [
+      IonicModule,
+      CommonModule,
+      FormsModule,
+      ExploreContainerComponentModule,
+      Tab1PageRoutingModule,
++     ScrollingModule
+    ],
+    declarations: [Tab1Page]
+  })
+  export class Tab1PageModule {}
+```
+
+With this added, we have access to the Virtual Scroller in the Tab1Page component.
+
+## Usage
+
+The CDK Virtual Scroller can be added to a component by add the `cdk-virtual-scroll-viewport` to a component's template.
+
+```html
+<ion-content>
+  <cdk-virtual-scroll-viewport> </cdk-virtual-scroll-viewport>
+</ion-content>
+```
+
+`cdk-virtual-scroll-viewport` becomes the root of our scrollable content and is responsible for recycling DOM nodes as they scroll out of view.
+
+The DOM nodes at this point can be any content needed for an app. The difference is that when we want to iterate over a collection, `*cdkVirtualFor` is used instead of `*ngFor`.
+
+```html
+<ion-content>
+  <cdk-virtual-scroll-viewport>
+    <ion-list>
+      <ion-item *cdkVirtualFor="let item of items">
+        <ion-avatar slot="start">
+          <img src="https://loremflickr.com/40/40" />
+        </ion-avatar>
+        <ion-label>
+          {{item }}
+        </ion-label>
+      </ion-item>
+    </ion-list>
+  </cdk-virtual-scroll-viewport>
+</ion-content>
+```
+
+Here, `items` is an array, but it can be an array, `Observable<Array>`, or `DataSource`. `DataSource` is an abstract class that can provide the data needed as well as utility methods. For more details, checkout the [CDK Virtual Scroller docs](https://material.angular.io/cdk/scrolling/overview).
```

Comment:
```suggestion
Here, `items` is an array, but it can be an array, `Observable<Array>`, or `DataSource`. `DataSource` is an abstract class that can provide the data needed as well as utility methods. For more details, check out the [CDK Virtual Scrolling docs](https://material.angular.io/cdk/scrolling/overview).
```